addons/website_forum/controllers/website_forum.py

```diff
@@ -36,6 +36,21 @@ def _prepare_user_values(self, **kwargs):
             values['forum'] = kwargs.get('forum')
         elif kwargs.get('forum_id'):
             values['forum'] = request.env['forum.forum'].browse(int(kwargs.pop('forum_id')))
+        forum = values.get('forum')
+        if forum and forum is not True and request.uid != request.env.ref('base.public_user').id:
```

Comment:
Yes, but request.uid is not a user, right?
So it's a matter o comparing the costs of `request.uid != request.env.ref('base.public_user').id:` vs `self.env['res.users').browse(request.uid)._is_public`. This might depend on the ORM cache values in both cases...

Reply:
you can do `request.env.user`, no need to browse, and it's ORM cached so it should be ok